src/controllers.cpp

```diff
@@ -0,0 +1,276 @@
+/*
+ * Unpublished Copyright (c) 2009-2021 AutonomouStuff, LLC, All Rights Reserved.
+ *
+ * This file is part of the PACMod ROS 1.0 driver which is released under the MIT license.
+ * See file LICENSE included with this software or go to https://opensource.org/licenses/MIT for full license details.
+ */
+
+#include "pacmod_game_control/controllers.h"
+
+#include <pacmod_msgs/SystemCmdInt.h>
+
+// --- Generic gamepad controller (Logitech F310, XBOX)
+Controller::Controller()
+{
+  axes_[JoyAxis::LEFT_STICK_LR] = 0;
+  axes_[JoyAxis::LEFT_STICK_UD] = 1;
+
+  axes_[JoyAxis::RIGHT_STICK_LR] = 3;
+  axes_[JoyAxis::RIGHT_STICK_UD] = 4;
+
+  axes_[JoyAxis::LEFT_TRIGGER_AXIS] = 2;
+  axes_[JoyAxis::RIGHT_TRIGGER_AXIS] = 5;
+
+  axes_[JoyAxis::DPAD_LR] = 6;
+  axes_[JoyAxis::DPAD_UD] = 7;
+
+  btns_[JoyButton::BOTTOM_BTN] = 0;
+  btns_[JoyButton::RIGHT_BTN] = 1;
+  btns_[JoyButton::LEFT_BTN] = 2;
+  btns_[JoyButton::TOP_BTN] = 3;
+
+  btns_[JoyButton::LEFT_BUMPER] = 4;
+  btns_[JoyButton::RIGHT_BUMPER] = 5;
+
+  btns_[JoyButton::BACK_SELECT_MINUS] = 6;
+  btns_[JoyButton::START_PLUS] = 7;
+
+  // Currently unused
+  btns_[JoyButton::LEFT_STICK_PUSH] = 9;
+  btns_[JoyButton::RIGHT_STICK_PUSH] = 10;
+}
+
+void Controller::set_controller_input(const sensor_msgs::Joy& joy_msg)
+{
+  input_msg_ = joy_msg;
+}
+
+float Controller::get_accelerator_value()
+{
+  // The triggers report 1.0 when untouched, and change negatively all the way to -1.0 when fully pressed
+  return -(input_msg_.axes[axes_[JoyAxis::RIGHT_TRIGGER_AXIS]] - 1.0) / 2;
+}
+
+float Controller::get_brake_value()
+{
+  // The triggers report 1.0 when untouched, and change negatively all the way to -1.0 when fully pressed
+  return -(input_msg_.axes[axes_[JoyAxis::LEFT_TRIGGER_AXIS]] - 1.0) / 2.0;
+}
+
+float Controller::get_steering_value(JoyAxis steering_axis)
```

Comment:
What is the reason to have an input argument to get steering value?

Reply:
This can be removed once this other pr is merged: https://github.com/astuff/pacmod_game_control/pull/91